### ISSUE TITLE
It re-downloaded torch.sigmoid when running a script each time

### ISSUE THREAD

**Issue by CroissantAuBeurre** [pytorch/pytorch#42854]
## 🚀 Feature

When running a project that was developed a year ago, the sigmoid function was implemented from `nn.functional.sigmoid`, so, it flagged the below message and downloaded `torch.sigmoid` automatically. It worked well and the script ran smoothly.

Yet, when running the script once again, it re-downloaded `torch.sigmoid` again, such configuration
is a bit tedious since it had been downloaded earlier.

```shell
UserWarning: nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.
  warnings.warn("nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.")
  1%|▊                                                                              | 4/419 [00:12<22:17,  3.22s/it]
```

## Motivation

Sometimes, for those old projects, it would be safer to keep the old functions in the short term, rather than updating that function immediately. 

So, is it possible to keep both modes (auto/manual) to adapt to the different scenarios?

## Pitch

It'd better flag the warning message, rather than downloading it automatically, in particular, not downloading it again and again when running the same script.
